containers/src/Kokkos_UnorderedMap.hpp

```diff
@@ -119,6 +119,28 @@ class UnorderedMapInsertResult {
   uint32_t m_status;
 };
 
+/// \class UnorderedMapInsertOpTypes
+/// \brief Operations applied to the values array upon subsequent insertions.
+/// The default behavior when a k,v pair already exists in the UnorderedMap is
+/// to perform no operation. Alternatively, the caller may select to
+/// instantiate the UnorderedMap with the AtomicAdd insert operator such that
+/// duplicate keys accumulate values into the given values array entry.
+template <class ValueTypeView = int, class ValuesIdxType = uint32_t,
+          class ValueType = int>
```

Comment:
Do you need both `ValueTypeView` and `ValueType`? AFAICT, `impl_value_type` is `typename value_type_view::non_const_value_type` in the implementation of `UnorderedMap`, right?

Reply:
Thanks for all the helpful feedback, @masterleinad. I think `ValueTypeView` is required since, if the user wants to specify a insert op prior to defining the `map_type`, UnorderedMapInsertOpTypes must be defined before `map_type`. However, perhaps there is a more elegant way to achieve this same result with templates?